org.eclipse.m2e.pde/src/org/eclipse/m2e/pde/MavenTargetLocation.java

```diff
@@ -273,6 +274,10 @@ private boolean isPomType(Artifact artifact) {
 		return POM_PACKAGE_TYPE.equals(artifact.getExtension());
 	}
 
+	private boolean isFeature(Artifact artifact) {
+		return new FeatureParser().parse(artifact.getFile()) != null;
```

Comment:
I think parsing the feature to check is a bit heavy here, if there is not other way at laest the parse feature infos should be returned and used to prevent additional parsing afterwards.

Reply:
Done. I now simply check, whether the artifact contains a "feature.xml" in the root directory without looking at its content.
The file is only parsed once and only if the artifact is an actual feature.